akd/src/storage/memory.rs

```diff
@@ -187,6 +202,38 @@ impl Storage for AsyncInMemoryDatabase {
         Ok(map)
     }
 
+    async fn tombstone_value_states(&self, keys: &[ValueStateKey]) -> Result<(), StorageError> {
```

Comment:
We could but that implies that all use cases for tombstone'ing is related to data retention. While that is "probably" accurate, there could be cases where you'd want to tombstone an intermediate value in non-key-transparency situations (i.e. We want the first and last value, and tombstone in the middle).

Additionally, constructing the `ValueStateKey`s is relatively simple if we add a storage layer contract where you can say select a collection of value states between epochs (e.g. `limited_key_history`). Does that all make sense?

Reply:
That makes sense! I was gunning for efficiency here and trying to avoid two round trips to the database (say with an `UPDATE ... WHERE...` query). But I think it's an OK tradeoff for the ability to tombstone specific `ValueStates`.